Tests/Normalizer/KebabKeysNormalizerTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Tests\Normalizer;
+
+use FOS\RestBundle\Normalizer\KebabKeysNormalizer;
+use PHPUnit\Framework\TestCase;
+
+class KebabKeysNormalizerTest extends TestCase
+{
+    /**
+     * @expectedException \FOS\RestBundle\Normalizer\Exception\NormalizationException
+     */
+    public function testNormalizeSameValueException()
+    {
+        $normalizer = new KebabKeysNormalizer();
+        $normalizer->normalize([
+            'foo' => [
+                'foo-bar' => 'foo',
+                'foo-Bar' => 'foo',
+            ],
+        ]);
+    }
+
+    /**
+     * @dataProvider normalizeProvider
+     */
+    public function testNormalize(array $array, array $expected)
+    {
+        $normalizer = new KebabKeysNormalizer();
+        $this->assertEquals($expected, $normalizer->normalize($array));
+    }
+
+    public function normalizeProvider()
+    {
+        $array = $this->normalizeProviderCommon();
+        $array[] = array(
+            array('_-username' => 'foo', '-password' => 'bar', '-foo-bar' => 'foobar'),
+            array('_Username' => 'foo', 'Password' => 'bar', 'FooBar' => 'foobar'),
+        );
+
+        return $array;
+    }
+
+    private function normalizeProviderCommon()
```

Comment:
what's the point of this method it's only used once?

Reply:
removed.